### PR TITLE
Update theme

### PR DESCRIPTION
**Pull request by srallen**
_Please request review from `@zooniverse/frontend` team. If PR is related to design, please request review from `@beckyrother` in addition._ 

Package: several

Updates the theme with the changes requested by and approved by Becky. This will be a major version update. WIP because of incoming typography updates.

# Review Checklist

## General

- [x] Are the tests passing locally and on Travis?
- [x] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [x] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [x] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [x] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Comment by srallen** [comment]
Tests are failing because the theme context extender seems to be failing to deeply merge the themes for some reason... Investigating.

**Comment by srallen** [comment]
Some of the test failures were actually tests that specifically used enzyme's `render`. To quickly fix these, I just converted them to use shallow rendering instead as that's a bit more consistent with the rest of our test suite. 

Another set of failures were because of the use of `mount` and a child component somewhere now using the theme, so the theme had to be injected by way of a wrapping component/prop.

**Comment by srallen** [comment]
I could use everyone's help in reviewing this to make sure there aren't any style regressions, particularly having to do with colors. Thanks!

**Comment by srallen** [comment]
One change to note:
There is a typography theme applied to the markdown now. The updated styles for the headers ended up having an H5 that looked like an anchor link, so to distinguish them, I permanently gave the anchor a text underline and then bolded it on hover so it has a hover style. This is maybe something to review with the new designer as I didn't realize this conflict would arise and didn't have a chance to review it with Becky. I'm going to remove the bolding on hover because I see now in practice, it can cause an awkward shift sometimes. Notably in the subject metadata modal. Hopefully the cursor change is sufficient.

File: packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/DoneButton/DoneButton.js [hunk 1]
```diff
@@ -1,37 +1,28 @@
...
   completed = false,
   disabled = false,
-  hasNextStep = false,
   onClick = () => true
 }) {
   if (!completed) {
     return (
-      <Button
+      <PrimaryButton
+        color='green'
```
Comment (mcbouslog) [inline packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/DoneButton/DoneButton.js:21]: Should this be "neutral-3" or "status-ok" (both are #1ED359)?

**Review by mcbouslog — APPROVED** [review]
LGTM. Noting a few changes (not all changes, as many were related to change to/of "neutral-1" as expected) from current master to update. All of the following, per code or comments, I think are correct, but can't hurt to confirm:

light theme
- "Done" and "Done & Talk" font color from white to black
- "Done" button from darker green (#078F52, old "status-ok") to lighter green (#1ED359, new "status-ok")
- Project Home, about section, h1 from black to #005D69 "neutral-1" (per new Markdownz theme)
- Project Home, Connect With text from lighter shade of teal to darker shade of teal (#005D69, "neutral-1"), text spacing increase

dark theme
- "Done" and "Done and Talk" focus/hover from lighter shading to light blue and light green (#1ED359, new "status-ok") consistent with light theme
- Project Home, about section, h1 from white to "accent-1" (per new Markdownz theme)

**Review by goplayoutside3 — APPROVED** [review]
This PR looks good to me! I reviewed the pages I’m most familiar with, including the Project /about pages. I also manually imported the new theme.js to `mapping-viz-tools` which uses @zooniverse/grommet-theme and the styling looks good there too!

File: packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/DoneButton/DoneButton.js (hunk 1 above)
Comment (srallen) [reply packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/DoneButton/DoneButton.js:21]: The API for the `PrimaryButton` actually takes just simple color names. The prop `green` here will map to the correct theme color variable. We could consider changing this for consistency in the future maybe?

Comment (mcbouslog) [reply packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/components/DoneButton/DoneButton.js:21]: Yep def not blocking, consider in the future sounds good to me.

**Pull request merged**